libs/ledger-live-common/src/api/Evm.ts

```diff
@@ -0,0 +1,213 @@
+import { ethers } from "ethers";
+import BigNumber from "bignumber.js";
+import { Operation } from "@ledgerhq/types-live";
+import { CryptoCurrency } from "@ledgerhq/types-cryptoassets";
+import {
+  EtherscanOperation,
+  Transaction as EvmTransaction,
+} from "../families/evm/types";
+import { transactionToEthersTransaction } from "../families/evm/transaction";
+import { makeLRUCache } from "../cache";
+import network from "../network";
+import {
+  etherscanOperationToOperation,
+  scanApiForCurrency,
+} from "../families/evm/logic";
+
+type AsyncApiFunction = (api: ethers.providers.JsonRpcProvider) => Promise<any>;
+
+/**
+ * Connects to RPC Node
+ */
+async function withApi(
+  currency: CryptoCurrency,
+  execute: AsyncApiFunction
+): Promise<any> {
```

Comment:
Would be nice if we can define the return type !

Reply:
By definition we can't, withApi is a wrapper to all API methods in order to provide the necessary objects to all the other methods below, which means the return can be anything, it's really going to be defined by the return type of the `execute` method that we're setting as a parameter.

Reply:
Not nice then :disappointed:

Reply:
Well that's actually the only time it really makes sense to have `any` 😛 
`withApi` is never used by itself or exiting the scope of this file (that's why it's not exported), it's always wrapping an `execute` method that will have a clear return type 🤷‍♂️

Reply:
Ok I changed it for a templatized version, just for your beautiful 👀